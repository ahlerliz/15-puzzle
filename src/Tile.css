.Tile {
  height: 100px;
  width: 100px;
  border-radius: 8px;
  background-color: rgb(94, 49, 37);
  transition: background-color 0.5s ease;
  cursor: pointer;
  color:black;
  font-weight: 700;
  font-size: xx-large;
  background-image: url("wooden-tile.jpeg")
}

.Tile-empty {
  background-color: rgb(0, 0, 0);
  background-image: none;
}