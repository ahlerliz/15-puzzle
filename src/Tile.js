import React from "react";
import "./Tile.css";

/** A single tile on the board.
 *
 * This has no state --- just two props:
 *
 * - moveTilesHere: a function rec'd from the board which moves tiles 
 *        based on the clicked tile
 *
 * - value: number 1-15 or null
 *
 * This handles clicks --- by calling moveTilesHere
 *
 **/

function Tile({ moveTilesHere, value }) {
  const classes = `Tile ${value ? "" : "Tile-empty"}`;
  return <td className={classes} onClick={moveTilesHere} role="button" >{value}</td>;
}

export default Tile;
