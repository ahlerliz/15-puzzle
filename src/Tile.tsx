import React from "react";
import "./Tile.css";

/** A single tile on the board.
 *
 * This has no state --- just two props:
 *
 * - moveTilesHere: a function rec'd from the board which moves tiles 
 *        based on the clicked tile
 *
 * - value: number 1-15 or null
 *
 * This handles clicks --- by calling moveTilesHere
 *
 **/
interface Props {
  moveTilesHere: () => void,
  value: number | null | undefined;
}

function Tile({ moveTilesHere, value }: Props) {
  const classes = `Tile ${value ? "" : "Tile-empty"}`;
  return <td className={classes} onClick={moveTilesHere} role="button" >{value}</td>;
}

export default Tile;
