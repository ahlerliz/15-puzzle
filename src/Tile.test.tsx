import React from "react";
import { render } from "@testing-library/react";
import Tile from "./Tile";

describe("<Tile /> rendering", function () {
  let container: any;

  beforeEach(function () {
    // add a TR to the document created by the test
    // to avoid warnings in the test output
    // about appending a TD to a DIV
    let tr = document.createElement("tr");
    container = document.body.appendChild(tr);
  });

  it("renders without crashing", function () {
    render(<Tile value/>, { container });
  });

  it("matches snapshot when numbered", function () {
    const { asFragment } = render(<Tile value />, { container });
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when empty", function () {
    const { asFragment } = render(<Tile value/>, { container });
    expect(asFragment()).toMatchSnapshot();
  });
});
