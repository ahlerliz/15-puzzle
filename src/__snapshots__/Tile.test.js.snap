// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> rendering matches snapshot when empty 1`] = `<DocumentFragment />`;

exports[`<Tile /> rendering matches snapshot when numbered 1`] = `<DocumentFragment />`;
